teuthology/task/internal.py

```diff
@@ -159,6 +159,9 @@ def lock_machines(ctx, config):
             break
         elif not ctx.block:
             assert 0, 'not enough machines are available'
+        else:
+            how_many = how_many - len(newly_locked)
+            assert how_many > 0
```

Comment:
(the only reason the assert is there is that I could think of no good way to properly test this without inventing a lot more mechanism.)